app/code/Magento/Checkout/Model/ShippingInformationManagement.php

```diff
@@ -199,26 +214,39 @@ public function saveAddressInformation(
             );
         }
 
-        /** @var \Magento\Checkout\Api\Data\PaymentDetailsInterface $paymentDetails */
+        /** @var PaymentDetailsInterface $paymentDetails */
         $paymentDetails = $this->paymentDetailsFactory->create();
         $paymentDetails->setPaymentMethods($this->paymentMethodManagement->getList($cartId));
         $paymentDetails->setTotals($this->cartTotalsRepository->get($cartId));
         return $paymentDetails;
     }
 
+    /**
+     * Validate shipping address
+     *
+     * @param AddressInterface $address
+     * @return void
+     * @throws StateException
+     */
+    private function validateAddress(AddressInterface $address): void
+    {
+        if (!$address || !$address->getCountryId()) {
```

Comment:
`$address` cannot be null with Type hint.

Reply:
looks like it's the bug that it's not optional here